src/cost_estimates/pessimistic.rs

```diff
@@ -0,0 +1,336 @@
+        conn.query_row(sql, &[identifier], |row| row.get(0))
+            .optional()
+            .expect("SQLite failure")
+            .unwrap_or_else(|| Samples { items: vec![] })
+    }
+
+    fn get_estimate_sqlite(conn: &Connection, identifier: &str) -> Option<u64> {
+        let sql = "SELECT current_value FROM pessimistic_estimator WHERE estimate_key = ?";
+        conn.query_row::<i64, _, _>(sql, &[identifier], |row| row.get(0))
+            .optional()
+            .expect("SQLite failure")
+            .map(|x_i64| {
+                u64::try_from(x_i64).expect("DB corrupt, non-u64-valid estimate was stored")
+            })
+    }
+}
+
+impl PessimisticEstimator {
+    pub fn open(p: &Path, log_error: bool) -> Result<PessimisticEstimator, EstimatorError> {
+        let db =
+            sqlite_open(p, rusqlite::OpenFlags::SQLITE_OPEN_READ_WRITE, false).or_else(|e| {
+                if let SqliteError::SqliteFailure(ref internal, _) = e {
+                    if let rusqlite::ErrorCode::CannotOpen = internal.code {
+                        let mut db = sqlite_open(
+                            p,
+                            rusqlite::OpenFlags::SQLITE_OPEN_CREATE
+                                | rusqlite::OpenFlags::SQLITE_OPEN_READ_WRITE,
+                            false,
+                        )?;
+                        let tx = tx_begin_immediate_sqlite(&mut db)?;
+                        PessimisticEstimator::instantiate_db(&tx)?;
+                        tx.commit()?;
+                        Ok(db)
+                    } else {
+                        Err(e)
+                    }
+                } else {
+                    Err(e)
+                }
+            })?;
+
+        Ok(PessimisticEstimator { db, log_error })
+    }
+
+    /// Check if the SQL database was already created. Necessary to avoid races if
+    ///  different threads open an estimator at the same time.
+    fn db_already_instantiated(tx: &SqliteTransaction) -> Result<bool, SqliteError> {
+        table_exists(tx, "pessimistic_estimator")
+    }
+
+    fn instantiate_db(tx: &SqliteTransaction) -> Result<(), SqliteError> {
+        if !Self::db_already_instantiated(tx)? {
+            tx.execute(CREATE_TABLE, rusqlite::NO_PARAMS)?;
+        }
+
+        Ok(())
+    }
+
+    fn get_estimate_key(
+        tx: &TransactionPayload,
+        field: &CostField,
+        evaluated_epoch: &StacksEpochId,
+    ) -> String {
+        let tx_descriptor = match tx {
+            TransactionPayload::TokenTransfer(..) => "stx-transfer".to_string(),
+            TransactionPayload::ContractCall(cc) => {
+                // Epoch key is "" before Epoch2_05 for backwards compatibility.
+                let epoch_marker = match evaluated_epoch {
+                    StacksEpochId::Epoch10 => "",
+                    StacksEpochId::Epoch20 => "",
+                    StacksEpochId::Epoch2_05 => ":2.05",
+                    // Note: Reuse 2.05 in 2.1.
```

Comment:
This should be `:2.1`.  Estimates reset across epochs.

Reply:
We'll be adding new cost functions for new Clarity methods, and I can't promise that we won't change any existing cost functions.

Reply:
guess it can't hurt much to make its own key... except for the fact that estimates will reset for a brief period.